### PR TITLE
feat: refactor auth action

### PR DESCRIPTION
**Pull request by jannyHou**
### Description

connect to https://github.com/strongloop/loopback-next/issues/2467

This PR starts to refactor the authentication action to takes in a getter for the newly created AuthenticationStrategy interface in https://github.com/strongloop/loopback-next/pull/2688.

- The new strategy resolver should be agnostic of a strategy is passport based or not.
- The `StrategyAdapter` is also refactored to implement the `AuthenticationStrategy` interface
- Having the PR as a draft to prove the interface in https://github.com/strongloop/loopback-next/pull/2688 is good enough to describe a universal auth strategy. Will turn it to a formal one to finish story #2467 

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [x] `npm test` passes on your machine
- [x] New tests added or existing tests modified to cover all changes
- [x] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: .vscode/settings.json [hunk 1]
```diff
@@ -1,5 +1,7 @@
 {
-  "editor.rulers": [80],
+  "editor.rulers": [
+    80
```
Comment (jannyHou) [inline .vscode/settings.json]: will revert it.

File: packages/authentication/src/types.ts [hunk 2]
```diff
@@ -22,3 +22,9 @@ export interface UserProfile {
   name?: string;
   email?: string;
 }
+
+export interface AuthenticationStrategy {
+  name: string;
+  isPassportStrategy?: boolean;
```
Comment (bajtos) [inline packages/authentication/src/types.ts]: -1

AFAIK, passport strategies use different API contract than LB4 `AuthenticationStrategy`. IMO, we should tell people to use `StrategyAdapter` to convert Passport strategies into LB strategies.

Comment (jannyHou) [reply packages/authentication/src/types.ts]: @bajtos yeah that flag is added at the wrong place, see my comment https://github.com/strongloop/loopback-next/commit/53cfeebbaf2ebe2897e28f51432f6fb5d1a4e946#r33223459 .

**Comment by jannyHou** [comment]
Close this one in favor of https://github.com/strongloop/loopback-next/pull/2822

**Pull request closed without merging**